cpp/src/trajectory/derive_trajectories.cu

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <rmm/thrust_rmm_allocator.h>
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/groupby.hpp>
+#include <cudf/sorting.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/table/table_view.hpp>
+#include <memory>
+#include <vector>
+
+#include "trajectory.hpp"
+
+namespace cuspatial {
+namespace experimental {
+namespace detail {
+
+std::unique_ptr<cudf::experimental::table> derive_trajectories(
+    cudf::column_view const& id, rmm::mr::device_memory_resource* mr,
+    cudaStream_t stream) {
+  using typename cudf::column;
+  using typename cudf::table_view;
+  using namespace cudf::experimental;
+
+  std::vector<groupby::aggregation_request> reqs{};
+  reqs.reserve(2);
+
+  // append count ids aggregation
+  reqs.push_back(groupby::aggregation_request{id});
+  reqs[0].aggregations.push_back(make_count_aggregation());
+
+  // append nth_element aggregation to force method=sort
+  reqs.push_back(groupby::aggregation_request{id});
+  reqs[1].aggregations.push_back(make_nth_element_aggregation(0));
+
+  // do the needful
```

Comment:
// add a helpful...

... comment, please. :)

Reply:
Just having a bit of fun with one of my genuinely favorite phrases :smile: :
> [Do the needful](https://en.wikipedia.org/wiki/Indian_English#Vocabulary): To do that which is necessary or required, with the respectful implication that the other party is trusted to understand what needs doing without being given detailed instruction